src/jobs/deploy-s3.yml

```diff
@@ -1,9 +1,80 @@
-description: >
+description: |
   This job deploys an application to aws s3.
 
 executor: node/default
 
+parameters:
+  source:
+    type: string
+    default: SOURCE
+    description: |
+      a local directory path to sync with s3
+
+  s3_bucket:
+    type: string
+    description: |
+      A URI to an S3 bucket
+
+  tag:
+    type: string
+    description: |
+      tag for the repo
+
+  aws_region:
+    type: env_var_name
+    default: AWS_REGION_N_VA
+    description: |
+      Specify the region for aws
+
+  aws_access_key_id:
+    default: AWS_ACCESS_KEY_ID
+    type: env_var_name
+    description: |
+      AWS access key id for IAM role. Set this to the name of
+      the environment variable you will use to hold this
+      value, i.e. AWS_ACCESS_KEY.
+
+  aws_secret_access_key:
+    default: AWS_SECRET_ACCESS_KEY
+    type: env_var_name
+    description: |
+      Secret Key for MACHINE_AWS_ECS
+
+  mentions:
+    type: env_var_name
+    default: MENTIONS
+    description: |
+      MENTIONS is a the environment variable name for the slack context.
+
 steps:
+  - checkout
+  - tag-repo:
+      tag: <<parameters.tag>>
+  - run:
+      name: Install dependencies
+      command: npm install
   - run:
-      name: "Not Yet Implemented"
-      command: echo "Not Yet Implemented"
+      name: Build Application
+      command: npm ng build --prod --no-progress
+  - setup-aws:
```

Comment:
We can skip this step and attach the aws-access-key-id and aws-secret-access-key to the aws-s3/sync request because it handles aws cli setup for us.

Reply:
[aws-s3/sync](https://circleci.com/developer/orbs/orb/circleci/aws-s3#commands-sync)
[aws-cli/setup](https://circleci.com/developer/orbs/orb/circleci/aws-cli#commands-setup)

If you checkout the source code on line 32, they do the setup command. And on line 74 of aws-cli/setup they install the cli